R/clean_variable_spelling.R

```diff
@@ -211,9 +234,52 @@ clean_variable_spelling <- function(x = data.frame(), wordlists = list(), spelli
     wordlists    <- wordlists[names(wordlists) != ".global"]
     has_global   <- !is.null(global_words)
     # Iterate over the names of the dictionaries -----------
-    to_iterate <- intersect(names(wordlists), names(x))
+    
+    
+    # extract vars to check from wordlists (both plain and regex)
+    vars_check <- names(wordlists)
+    is_var_regex <- grepl("^\\.regex", vars_check)
+    vars_check_plain <- vars_check[!is_var_regex]
+    vars_check_regex <- vars_check[is_var_regex]
+    vars_check_regex_extract <- gsub("\\.regex[[:space:]]", "", vars_check_regex)
+    
+    # which cols in x match each regex var (returns 1 list element for each .regex key)
+    vars_regex_match_list <- lapply(
+      vars_check_regex_extract,
+      function(j) names(x)[grepl(j, names(x), perl = TRUE)]
```

Comment:
This will have the same effect

```suggestion
      FUN = grepl, 
      x = names(x), value = TRUE, perl = TRUE)
```